pennylane/measurements/measurements.py

```diff
@@ -299,11 +303,16 @@ def __repr__(self):
         return f"{self.obs}"
 
     def __copy__(self):
-        return self.__class__(
-            obs=copy.copy(self.obs),
-            wires=self._wires,
-            eigvals=self._eigvals,
-        )
+        cls = self.__class__
```

Comment:
The way works with the different call signatures for child classes and copies over attributes like `all_outcomes` or `log_base` when they exist.

Reply:
as long as no-one adds anything private (ie. a property with a dunder name), using `vars(self)` should be fine

Reply:
```
>>> vars(ExpectationMP(eigvals=(-1,1), wires=qml.wires.Wires(0)))
{'obs': None,
 'id': None,
 '_wires': <Wires = [0]>,
 '_eigvals': array([-1,  1]),
 'name': 'Identity',
 'data': []}
```

copies private variables as well.  Got me worried for a second about `_eigvals`.

Reply:
If they were doing something like that, we would expect them to do enough testing and be able to look at the source code for this method.